rust/routee-compass-powertrain/src/routee/prediction/interpolation/utils.rs

```diff
@@ -0,0 +1,109 @@
+use ordered_float::OrderedFloat;
+
+pub fn linspace(x0: f64, xend: f64, n: usize) -> Vec<f64> {
```

Comment:
wondering if you wrote these functions or if you grabbed them from somewhere. they look like something from rosetta code? if so, we should attribute them.

Reply:
yeah good call, that one was written using an [SO answer](https://stackoverflow.com/a/70840233) as a starting point. I'll add attribution in for it